### ISSUE TITLE
org-capture-journal-file is not expanded with org-directory

### ISSUE THREAD

**Issue by game-of-drones** [hlissner/doom-emacs#2343]
**What did you expect to happen?**
I added `(setq org-directory "/path/to/my/org")` in `.doom/config.el` and run `~/.emacs.d/bin/doom refresh`. After opening emacs and press
```SPC X j```
I want it to capture a journal entry in `/path/to/my/org/journal.org`

**What actually happened?**
I got a message saying "Heading not found on level 1: Inbox". And when I list ibuffer with `SPC b I`, I found a new file is created `mycurrentdir/journal.org`, where `mycurrentdir` is the path where I started emacs. I.e., it didn't expand "journal.org" with org-directory.

What's weird is that if I capture a todo by `SPC X t`, it open the correct file to add entry which is `/path/to/my/org/todo.org`. capture notes also works. Only journal doesn't work.

**Additional details:**
1. When I grep "org-capture-todo-file", I can see following lines  in `modules/lang/org/autoload/org-capture.el`
```
79:(defun +org-capture-todo-file ()
80:  "Expand `+org-capture-todo-file' from `org-directory'.
81:If it is an absolute path return `+org-capture-todo-file' verbatim."
82:  (expand-file-name +org-capture-todo-file org-directory))
```

2. When I grep "org-capture-notes-file", similar expand-file-name is called

3. However, when I grep "org-capture-journal-file", I don't see such expand-file-name called.

So my guess is that in ".emacs.d/modules/lang/org/config.el", something needs to be done for org-capture-journal-file to expand it correctly.

Thanks!

**Comment by game-of-drones** [comment]
** Update
I did the following settings in `.doom.d/config.el`. Now `SPC X j` can find the correct file `~/Documents/jrnl/my_journal.org`. 
```
;; If you intend to use org, it is recommended you change this!
(setq org-directory "~/Documents/jrnl")
(after! org
  (setq +org-capture-journal-file
        (expand-file-name "my_journal.org"  org-directory)
  )
)
```

However, it still complains `Heading not found on Level 1: Inbox`.  This seems to be related to the fact that in variable `org-capture-templates`, the entry for journal is like this.
```
 ("j" "Journal" entry
  (file+olp+datetree +org-capture-journal-file "Inbox")
```

**Comment by game-of-drones** [comment]
** Update 2
After the following setting in `.doom.d/config.el`, I added an entry for journal without the "Inbox" thing. Now it works as I wanted. Ideally I wanted to **modify** the existing entry for journal (without modifying doom's config), but I don't know how. So I have to add a new entry with shortcut "k", which is kind of awkward.
```
(setq org-directory "~/Documents/jrnl")
(after! org
  (setq +org-capture-journal-file
        (expand-file-name "my_journal.org"  org-directory)
  )

  (add-to-list 'org-capture-templates
               '("k" "Journal" entry (file+olp+datetree +org-capture-journal-file) "* %T %?\n%i\n%a" :prepend t)
  )
)
```

**Comment by harritaylor** [comment]
@game-of-drones I've been facing this issue for the past few days, thanks for finding the problem and fixing it! I've submitted a PR to address this (although not sure if it's the best way of solving the problem, we'll see) 
If you want a quick fix in the meantime, just add the code you wrote (described in [this commit](https://github.com/hlissner/doom-emacs/pull/2360/commits/2fcb74dead186fec52bcc0a597773ce81b30336b)) to line 270 at ~/.emacs.d/modules/lang/org/config.el.

**Comment by alx** [comment]
Hi all,

I had this issue, and find this place after a google search.

My way to resolve the "Heading not found on level 1: Inbox" was to add a `* INBOX` line in `~/org/journal.org` file.

I'm new to emacs and `doom emacs`, it might not be the correct answer, but it worked for me.

Thanks :)

**Comment by hlissner** [comment]
Just a heads up: this was fixed last week in 786dec3e504e; the journal capture template no longer targets the "inbox".
